internal/controller/datadogagent/global/dependencies.go

```diff
@@ -0,0 +1,215 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package global
+
+import (
+	"fmt"
+
+	"github.com/go-logr/logr"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/DataDog/datadog-operator/api/datadoghq/v2alpha1"
+	apiutils "github.com/DataDog/datadog-operator/api/utils"
+	"github.com/DataDog/datadog-operator/internal/controller/datadogagent/common"
+	"github.com/DataDog/datadog-operator/internal/controller/datadogagent/component/objects"
+	"github.com/DataDog/datadog-operator/internal/controller/datadogagent/feature"
+	"github.com/DataDog/datadog-operator/pkg/constants"
+	"github.com/DataDog/datadog-operator/pkg/controller/utils/comparison"
+	"github.com/DataDog/datadog-operator/pkg/kubernetes/rbac"
+	"github.com/DataDog/datadog-operator/pkg/secrets"
+)
+
+func dependencies(logger logr.Logger, dda *v2alpha1.DatadogAgent, manager feature.ResourceManagers, componentName v2alpha1.ComponentName) []error {
+	var errs []error
+	// Credentials
+	if err := credentialDependencies(logger, dda, manager); err != nil {
+		errs = append(errs, err)
+	}
+
+	// DCA token
+	if err := dcaTokenDependencies(logger, dda, manager); err != nil {
+		errs = append(errs, err)
+	}
+
+	// Network policy
+	if err := networkPolicyDependencies(dda, manager, componentName); err != nil {
+		errs = append(errs, err)
+	}
+
+	// Secret backend
+	if err := secretBackendDependencies(logger, dda, manager, componentName); err != nil {
+		errs = append(errs, err)
+	}
+
+	return nil
+}
+
+func credentialDependencies(logger logr.Logger, dda *v2alpha1.DatadogAgent, manager feature.ResourceManagers) error {
```

Comment:
Credential and dca token dependencies are copied from the previous `global.go`. Network policy has a few small changes to use existing functions